### PR TITLE
Adds a Player picker.

### PR DESCRIPTION
**Pull request by utahclock**
picks random player and pulls up thier player panel, useful for when someone wants to give up thier antag status. and you want to replace that antag.
:cl:
rscadd: Admins can now randomly pick players.
/:cl:

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -946,3 +948,44 @@ var/list/admin_verbs_hideable = list(
...
+	var/choose_from_dead = input(src, "What group would you like to pick from?", "Selection", "Everyone") as null|anything in list("Everyone", "Living Only", "Dead Only")
+	if (!choose_from_dead)
+		return
+
+	var/list/player_pool = list()
+	for (var/mob/M in world)
+		if (!M.client || istype(M, /mob/new_player))
+			continue
+		if (what_group != "Everyone")
+			if ((what_group == "Traitors Only") && ROLE_TRAITOR)
```
Comment (Super3222) [inline code/modules/admin/admin_verbs.dm]: There are more than just traitors... what about replacing cultists or servants of ratvar?

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -946,3 +948,44 @@ var/list/admin_verbs_hideable = list(
...
+				continue
+			else if ((what_group == "Non-Traitors Only") && !ROLE_TRAITOR)
```
Comment (Super3222) [inline code/modules/admin/admin_verbs.dm]: If you change the above comment, than you'll have to change this to non antags.

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -946,3 +948,44 @@ var/list/admin_verbs_hideable = list(
...
+
+
+
+/client/proc/admin_pick_random_player()
+	set category = "Admin"
+	set name = "Pick Random Player"
+	set desc = "Picks a random logged-in player and brings up their player panel."
+
+
+	var/what_group = input(src, "What group would you like to pick from?", "Selection", "Everyone") as null|anything in list("Everyone", "Traitors Only", "Non-Traitors Only")
```
Comment (Super3222) [inline code/modules/admin/admin_verbs.dm]: I would've suggested letting someone pick what type of antag want, and having specific phrases for whether it'd go through (getting the message for who's in that group) or not.

File: code/modules/admin/admin_verbs.dm [hunk 4]
```diff
@@ -946,3 +948,44 @@ var/list/admin_verbs_hideable = list(
...
+	if (!what_group)
+		return
+	var/choose_from_dead = input(src, "What group would you like to pick from?", "Selection", "Everyone") as null|anything in list("Everyone", "Living Only", "Dead Only")
+	if (!choose_from_dead)
+		return
+
+	var/list/player_pool = list()
+	for (var/mob/M in world)
```
Comment (Super3222) [inline code/modules/admin/admin_verbs.dm:967]: You could use `for(var/mob/M in mob_list)`

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (utahclock) [reply code/modules/admin/admin_verbs.dm]: I can expand this I suppose.

File: code/modules/admin/admin_verbs.dm (hunk 3 above)
Comment (utahclock) [reply code/modules/admin/admin_verbs.dm]: This isnt just for making antags.. It for any role an admin might want to consider a random player for

**Comment by Alek2ander** [comment]
I suggest adding "non-antags who want to be %antag%" for every antag preference flag to the list.
Also maybe "players not picked for antag" - the remaining antag_candidates list in the ticker

File: code/modules/admin/admin_verbs.dm (hunk 4 above)
Comment (ShadowDeath6) [reply code/modules/admin/admin_verbs.dm:967]: mob_list only includes living, right?

Comment (Super3222) [reply code/modules/admin/admin_verbs.dm:967]: > mob_list only includes living, right?

@ShadowDeath6 You're thinking about living_mob_list.

**Comment by Super3222** [comment]
I like Alek2ander's suggestion, but it's better expand as a window with that, but it may overexceed this PR.

File: code/modules/admin/admin_verbs.dm (hunk 3 above)
Comment (monster860) [reply code/modules/admin/admin_verbs.dm]: Isnt this what `offer control to ghosts` is for

**Comment by utahclock** [comment]
No . as if you want to run event. But want yo pick random player.

**Comment by utahclock** [comment]
This picks both alive and dead players. Or if someone wants to give up
thier antag and you want to give it to random.

**Comment by utahclock** [comment]
Its been updated to pick from ALL antag types.

File: code/modules/admin/admin_verbs.dm (hunk 3 above)
Comment (utahclock) [reply code/modules/admin/admin_verbs.dm]: Monster no, This is so you can randomly pick a player, you can choose from living or dead. If you decide you want to give a role to someone .

Comment (utahclock) [reply code/modules/admin/admin_verbs.dm]: Antag or Non-antag.

**Pull request closed without merging**